src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -333,9 +333,16 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                 if (connectionStatus.Exception.Contains(nameof(DatabaseDoesNotExistException)))
                     DatabaseDoesNotExistException.ThrowWithMessage(_dbName, connectionStatus.Message);
             }
+
             if (connectionStatus.Type != SubscriptionConnectionServerMessage.MessageType.ConnectionStatus)
-                throw new Exception("Server returned illegal type message when expecting connection status, was: " +
-                                    connectionStatus.Type);
+            {
+                var message = "Server returned illegal type message when expecting connection status, was: " + connectionStatus.Type;
+
+                if (connectionStatus.Type == SubscriptionConnectionServerMessage.MessageType.Error)
+                    message += $". Exception: {connectionStatus.Exception}";
+
+                throw new Exception(message);
```

Comment:
Can we throw a different exception here?

Reply:
I was a bit afraid that throwing different exception might be a braking change for someone's error handling.

Reply:
Anyone who is currently handling this already catches `Exception`, after all :-)

Reply:
I was thinking about using one of our existing `Subscription...Exception` but I will create a new one for that so it would be caught by generic `catch Exception()` for sure

Reply:
Handled in https://github.com/ravendb/ravendb/pull/13470/commits/e4e2a9e6c0327b1bb21ef49ce8adbad63c3d8e1b